controller_manager/controller_manager/controller_manager_services.py

```diff
@@ -28,15 +28,17 @@
 import rclpy
 
 
-def service_caller(node, service_name, service_type, request, service_timeout=10.0):
+def service_caller(node, service_name, service_type, request, service_timeout=0.0):
     cli = node.create_client(service_type, service_name)
 
-    if not cli.service_is_ready():
-        node.get_logger().debug(
-            f"waiting {service_timeout} seconds for service {service_name} to become available..."
-        )
-        if not cli.wait_for_service(service_timeout):
-            raise RuntimeError(f"Could not contact service {service_name}")
+    while not cli.service_is_ready():
+        node.get_logger().info(f"waiting for service {service_name} to become available...")
+        if service_timeout:
+            if not cli.wait_for_service(service_timeout):
+                node.get_logger().fatal(f"Could not contact service {service_name}")
+                raise RuntimeError(f"Could not contact service {service_name}")
```

Comment:
I think, catching the exception thrown here would be beneficial. Currently, the spawner dies with a stacktrace which in my opinion isn't the right thing to do for an application level error such as this:

```
$ ros2 run controller_manager spawner --controller-manager-timeout 2.0 foobar
[INFO 2024-08-11 22:42:22.692] [spawner_foobar]: waiting for service /controller_manager/list_controllers to become available...
[FATAL 2024-08-11 22:42:24.719] [spawner_foobar]: Could not contact service /controller_manager/list_controllers
Traceback (most recent call last):
  File "/home/mauch/checkout/ros2_control_rolling/colcon_ws/install/controller_manager/lib/controller_manager/spawner", line 33, in <module>
    sys.exit(load_entry_point('controller-manager==4.15.0', 'console_scripts', 'spawner')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mauch/checkout/ros2_control_rolling/colcon_ws/install/controller_manager/lib/python3.12/site-packages/controller_manager/spawner.py", line 214, in main
    if is_controller_loaded(node, controller_manager_name, controller_name, controller_manager_timeout):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mauch/checkout/ros2_control_rolling/colcon_ws/install/controller_manager/lib/python3.12/site-packages/controller_manager/spawner.py", line 85, in is_controller_loaded
    controllers = list_controllers(node, controller_manager, service_timeout).controller
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mauch/checkout/ros2_control_rolling/colcon_ws/install/controller_manager/lib/python3.12/site-packages/controller_manager/controller_manager_services.py", line 66, in list_controllers
    return service_caller(
           ^^^^^^^^^^^^^^^
  File "/home/mauch/checkout/ros2_control_rolling/colcon_ws/install/controller_manager/lib/python3.12/site-packages/controller_manager/controller_manager_services.py", line 39, in service_caller
    raise RuntimeError(f"Could not contact service {service_name}")
RuntimeError: Could not contact service /controller_manager/list_controllers
[ros2run]: Process exited with failure 1
```

The FATAL logging would have been enough at this point in my opinion. We could either directly `sys.exit` from the point where we log the error or raise a specialized exception that we can catch in the `main` function of the spawner. Since the `controller_manager_services` module serves more like a library in my opinion it should not `sys.exit` directly, but we should catch the exception in the application. Since I have been adding stuff to this on mu fork, I will add one commit implementing this. This can then either be cherry-picked or not :-)